h/services/settings.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from h.models import Setting
+
+
+class SettingsService(object):
+    """A service for fetching and manipulating settings."""
+
+    def __init__(self, session):
+        """
+        Create a new settings service.
+
+        :param session: the SQLAlchemy session object
+        """
+        self.session = session
+
+    def get(self, key):
+        setting = self._fetch(key)
+
+        if setting is None:
+            return None
+
+        return setting.value
+
+    def put(self, key, value):
+        setting = self._fetch(key)
+
+        if setting is None:
+            setting = Setting(key=key)
+
+        setting.value = value
+        self.session.add(setting)
+
+    def delete(self, key):
+        setting = self._fetch(key)
+
+        if setting is not None:
+            self.session.delete(setting)
+
+    def _fetch(self, key):
+        return self.session.query(Setting).get(key)
+
+
+def settings_factory(context, request):
```

Comment:
In the new `h.settings` package I think we can probably stick to one service per module, so what do you think about reducing stutter and calling these factory functions just `factory` so that the setup looks like:

```
config.register_service_factory('.settings.factory', name='settings')
```